lib/autoproj/cli/version.rb

```diff
@@ -0,0 +1,62 @@
+require 'autoproj/cli/base'
+require 'rubygems'
+
+module Autoproj
+module CLI
+    class Version < Base
+        # List the version of autoproj and optionally include
+        # the installed dependencies with information about the requirement
+        # and the actual used version
+        def run(args, options = Hash.new)
+            puts "autoproj version: #{Autoproj::VERSION}"
+            if options[:deps]
+                dependency = Gem::Deprecate.skip_during do
+                    Gem::Dependency.new "autoproj", Autoproj::VERSION
+                end
+                autoproj_spec = dependency.matching_specs
+                if !autoproj_spec.empty?
+                    known_dependencies = collect_dependencies(dependency)
+                    installed_deps = {}
+                    known_dependencies.flatten.uniq do |spec|
+                        installed_deps[spec.name] = spec.version
+                    end
+                    puts "    specified dependencies:"
+                    autoproj_spec.first.dependencies.each do |dep|
+                        puts "        #{dep}: #{installed_deps[dep.name] || 'n/a'}"
+                        installed_deps.delete(dep.name)
+                    end
+                    puts "    implicit dependencies:"
+                    installed_deps.keys.sort.each do |name|
+                        puts "        #{name}: #{installed_deps[name]}" unless name == "autoproj"
+                    end
+                else
+                    Autoproj.warn "Dependencies and their versions could not be retrieved -- is gem packager installed?"
```

Comment:
I don't think this is the right warning message anymore ... At this point, I would classify the error as "internal error". Autoproj is always bootstrapped using bundler/gem.